packages/core/src/__tests__/client.test.ts

```diff
@@ -46,5 +52,119 @@ describe('client', () => {
     })
     const result3 = await client.findAll(klass, {})
     expect(result3).toHaveLength(4)
+
+    const result4 = await client.findOne(klass, {})
+    expect(result4).toEqual(result3[0])
+  })
+
+  it('should create client with plugins', async () => {
+    const txFactory = new TxFactory(core.account.System)
+    const txes = genMinModel()
+
+    txes.push(
+      txFactory.createTxCreateDoc(
+        core.class.Class,
+        core.space.Model,
+        {
+          label: 'PluginConfiguration' as IntlString,
+          extends: core.class.Doc,
+          kind: ClassifierKind.CLASS,
+          domain: DOMAIN_MODEL
+        },
+        core.class.PluginConfiguration
+      )
+    )
+
+    async function connectPlugin (handler: (tx: Tx) => void): Promise<ClientConnection> {
+      const hierarchy = new Hierarchy()
+
+      for (const tx of txes) hierarchy.tx(tx)
+
+      const transactions = new TxDb(hierarchy)
+      const model = new ModelDb(hierarchy)
+      for (const tx of txes) {
+        await transactions.tx(tx)
+        await model.tx(tx)
+      }
+
+      async function findAll<T extends Doc> (_class: Ref<Class<T>>, query: DocumentQuery<T>): Promise<FindResult<T>> {
+        return await transactions.findAll(_class, query)
+      }
+
+      return {
+        findAll,
+        tx: async (tx: Tx): Promise<TxResult> => {
+          if (tx.objectSpace === core.space.Model) {
+            hierarchy.tx(tx)
+          }
+          const result = await Promise.all([transactions.tx(tx)])
+          return result[0]
+        },
+        close: async () => {},
+
+        loadChunk: async (domain: Domain, idx?: number) => ({
+          idx: -1,
+          index: -1,
+          docs: {},
+          finished: true,
+          digest: ''
+        }),
+        closeChunk: async (idx: number) => {},
+        loadDocs: async (domain: Domain, docs: Ref<Doc>[]) => [],
+        upload: async (domain: Domain, docs: Doc[]) => {},
+        clean: async (domain: Domain, docs: Ref<Doc>[]) => {}
+      }
+    }
+    const spyCreate = jest.spyOn(TxProcessor, 'createDoc2Doc')
+    const spyUpdate = jest.spyOn(TxProcessor, 'updateDoc2Doc')
+
+    const pluginData1 = {
+      pluginId: 'testPlugin1',
+      transactions: []
+    }
+    const txCreateDoc1 = txFactory.createTxCreateDoc(core.class.PluginConfiguration, core.space.Model, pluginData1)
+    txes.push(txCreateDoc1)
+    const client1 = new TxOperations(await createClient(connectPlugin, ['testPlugin1' as Plugin]), core.account.System)
+    const result1 = await client1.findAll(core.class.PluginConfiguration, {})
+
+    expect(result1).toHaveLength(1)
+    expect(TxProcessor.createDoc2Doc).toBeCalledTimes(95)
```

Comment:
Not good, since on every model update, we will need update this as well.

Reply:
Let's take different approach.